mypyc/irbuild/builder.py

```diff
@@ -570,21 +571,22 @@ def get_assignment_target(
                 self.error("Cannot assign to the first argument of classmethod", line)
             if lvalue.kind == LDEF:
                 if symbol not in self.symtables[-1]:
+                    if isinstance(symbol, Var) and not isinstance(symbol.type, DeletedType):
```

Comment:
There's two places where `DeletedType` can show up:

- `except ... as <variable>` — there's no way `variable` can have an inferred optional type
- shenanigans with `del <variable>` that I haven't been able to figure out

The first shouldn't pose a problem (ie. mask an optional inferred type), but I'm not entirely sure about `del`. PTAL.

Reply:
Here's a way to generate a deleted type:
```py
def f() -> None:
    if int():
        x = 0
    else:
        x = None
    del x
    reveal_type(x)  # Revealed type is "<Deleted 'x'>"
```

Reply:
Ah thank you, `del x` shouldn't pose a problem then. You can't have a deleted variable later inferred to be an optional type on assignment (because you can only delete already defined variables ... which would have a type already).